src/packages/@library/forms/Input/Input.svelte

```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
+  import Edit from "$lib/assets/edit.svelte";
+  import { SearchIcon } from "$lib/assets/icons";
+  import { PencilIcon } from "@library/icons";
+  import { createEventDispatcher } from "svelte";
+
+  export let type: "text" | "password" | "search" = "text";
+  export let placeholder = "placeholder";
+  export let height = "30px";
+  export let width = "auto";
+
+  /**
+   * input class
+   */
+  export let disabled = false;
+  let componentClass = "";
+  export { componentClass as class };
+
+  let componentStyle = "";
+  export { componentStyle as style };
+  /**
+   * input value
+   */
+  export let value = "";
+
+  /**
+   * border color
+   */
+  export let defaultBorderColor = "transparent";
+  export let hoveredBorderColor = "red";
```

Comment:
all the props which are expected should have JSDocs comment. Secondly, why would we export `defaultBorderColor`? By name itself it is a `default` color which will be applied if nothing else is given. Why would we want to override it?

Reply:
lets take 3 scenarios 
1. If I want an input with default border color set to grey
2. If I want an input with default border color set to blue
3. 2. If I want an input with default border color set to none